examples/kbleds.c

```diff
@@ -32,18 +31,16 @@ static char kbledstatus = 0;
  * the LEDs reflect the actual keyboard status).  To learn more on this,
  * please see file: drivers/tty/vt/keyboard.c, function setledstate().
  */
-
-static void my_timer_func(unsigned long ptr)
+static void my_timer_func(struct timer_list *unused)
 {
-    unsigned long *pstatus = (unsigned long *)ptr;
     struct tty_struct *t = vc_cons[fg_console].d->port.tty;
 
-    if (*pstatus == ALL_LEDS_ON)
-        *pstatus = RESTORE_LEDS;
+    if (kbledstatus == ALL_LEDS_ON)
+        kbledstatus = RESTORE_LEDS;
     else
-        *pstatus = ALL_LEDS_ON;
+        kbledstatus = ALL_LEDS_ON;
 
-    (my_driver->ops->ioctl)(t, KDSETLED, *pstatus);
+    (my_driver->ops->ioctl)(t, KDSETLED, (unsigned long)kbledstatus);
```

Comment:
How can declaring `kbledstatus` as a `unsigned long` type.

Reply:
The original one did the same thing:
```cpp
  timer_setup(&my_timer, (void *)&my_timer_func, (unsigned long)&kbledstatus);
  unsigned long *pstatus = (unsigned long *)ptr;
  (my_driver->ops->ioctl)(t, KDSETLED, *pstatus);
```
And the type of function `my_driver->ops->ioctl`, which is `vt_ioctl`, is:
```cpp
int vt_ioctl(struct tty_struct *tty,
	     unsigned int cmd, unsigned long arg)
```

Reply:
See if we can reduce the casting. Example code should be as short (and plain) as possible.